services/src/main/java/org/keycloak/userprofile/config/UPConfigUtils.java

```diff
@@ -127,6 +130,25 @@ private static void validate(KeycloakSession session, UPAttribute attributeConfi
         }
         if (attributeConfig.getRequired() != null) {
             validateRoles(attributeConfig.getRequired().getRoles(), "required.roles", errors, attributeName);
+            validateScopes(attributeConfig.getRequired().getScopes(), errors, session);
+        }
+        if (attributeConfig.getSelector() != null) {
+            validateScopes(attributeConfig.getSelector().getScopes(), errors, session);
+        }
+    }
+
+    private static void validateScopes(Set<String> scopes, List<String> errors, KeycloakSession session) {
+        if (scopes == null) {
+            return;
+        }
+
+        for (String scope : scopes) {
+            RealmModel realm = session.getContext().getRealm();
+            Stream<ClientScopeModel> realmScopes = realm.getClientScopesStream();
+
+            if (!realmScopes.anyMatch(cs -> cs.getName().equals(scope))) {
+                errors.add(new StringBuilder("Invalid scope ").append(scope).toString());
```

Comment:
Error message should be probably better, like in role validtion, to name exact attribute and field where the problem is